platform/linux-generic/odp_timer.c

```diff
@@ -872,7 +876,17 @@ void _odp_timer_run_inline(int dec)
 	}
 
 	/* Check the timer pools. */
+#if CONFIG_TIMER_PROFILE_INLINE
```

Comment:
Since CONFIG_TIMER_PROFILE_INLINE is build time constant, normal if / else can be used instead of #if / #else.  Optimizer will remove the unused branch, so there won't be extra code by default.

Benefit of using normal if-clause is that all code is always visible to the compiler and some other change later does not break the profile code from building.

Please change all "#if CONFIG_TIMER_PROFILE_INLINE" to if (CONFIG_TIMER_PROFILE_INLINE), and remove #if from timer_local_t struct.